rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/LocalInfrastructure.java

```diff
@@ -97,19 +97,19 @@ public String getDescription() {
 
     @Override
     public void acquireAllNodes() {
-        readLock.lock();
+        this.readLock.lock();
         try {
-            logger.info("Starting node acquisition. Acquired nodes=" + getAcquiredNodes() + ", Handled nodes=" +
-                        getHandledNodes());
-            if (getAcquiredNodes() < getHandledNodes()) {
-                int differenceBetweenHandledAndAcquiredNodes = getDifferenceBetweenHandledAndAcquiredNodes();
+            logger.info("Starting node acquisition. Acquired nodes=" + getNumberOfAcquiredNodesWithLock() +
+                        ", Handled nodes=" + getNumberOfHandledNodesWithLock());
+            if (getNumberOfAcquiredNodesWithLock() < getNumberOfHandledNodesWithLock()) {
+                int differenceBetweenHandledAndAcquiredNodes = getDifferenceBetweenNumberOfHandledAndAcquiredNodesWithLock();
```

Comment:
can you compute difference and then use in if condition?

Reply:
I changed so that I do not execute `getNumberOfAcquiredNodesWithLock` and `getNumberOfHandledNodesWithLock` twice.